src/dinitctl.cc

```diff
@@ -1813,6 +1886,70 @@ static int trigger_service(int socknum, cpbuffer_t &rbuffer, const char *service
     return 0;
 }
 
+static int signal_send(int socknum, cpbuffer_t &rbuffer, const char *service_name, int sig_num)
+{
+    using namespace std;
+
+    handle_t handle;
+
+    if (!load_service(socknum, rbuffer, service_name, &handle, nullptr, true)) {
+        return 1;
+    }
+
+    // Issue SIGNAL command.
+    auto m = membuf()
+            .append<char>(DINIT_CP_SIGNAL)
+            .append(sig_num)
+            .append(handle);
+    write_all_x(socknum, m);
+
+    wait_for_reply(rbuffer, socknum);
+    if (rbuffer[0] == DINIT_RP_NAK) {
+        cerr << "dinitctl: cannot send signal to service.\n";
+        return 1;
+    }
+    if (rbuffer[0] == DINIT_RP_SIGNAL_NOPID) {
+        cerr << "dinitctl: could not get vaild PID of service; service is not process, "
+        "service in wrong state." << endl;
+        return 1;
+    }
+    if (rbuffer[0] == DINIT_RP_SIGNAL_BADSIG) {
+        cerr << "dinitctl: provided signal was invalid.\n";
+        return 1;
+    }
+    if (rbuffer[0] == DINIT_RP_SIGNAL_KILLERR) {
+        cerr << "dinitctl: failed sending signal to service.\n";
+        return 1;
+    }
+    if (rbuffer[0] != DINIT_RP_ACK) {
+        cerr << "dinitctl: protocol error.\n";
+        return 1;
+    }
+    rbuffer.consume(1);
+    return 0;
+}
+
+static int signal_list()
+{
+    using namespace std;
+    using namespace dinit_load;
+    cerr << "dinitctl: The following signal names are defined for this platform:";
+    int i = 0;
+    for (const auto &signal: signal_to_int_map) {
+        if (i%4 == 0) {
```

Comment:
Please use style consistent with the rest of the code base, including spaces around operators, indentation.